modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,300 @@
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'            => 'Dahua DVR Auth Bypas Scanner',
+      'Version'         => '$Revision: 1 $',
+      'Description'     => 'Scans for Dahua-based DVRs and then grabs settings. Optionally resets a user\'s password and clears the device logs',
+      'Author'          => [
+        'Jake Reynolds - Depth Security', # Vulnerability Discoverer
+        'Tyler Bennett - Talos Infosec' # Metasploit Module
+      ],
+      'References'      => [
+        [ 'CVE', '2013-6117' ],
+        [ 'URL', 'https://depthsecurity.com/blog/dahua-dvr-authentication-bypass-cve-2013-6117' ]
+      ],
+      'License'         => MSF_LICENSE
+       )
+    deregister_options('RHOST')
+    register_options([
+      OptString.new('USERNAME', [true, 'A username to reset', '888888']),
+      OptString.new('PASSWORD', [true, 'A password to reset the user with', 'abc123']),
```

Comment:
It seems that `888888` is a default account, so this is a fine default, but for `PASSWORD`, IMO it should default to something random.  In other words, change this line to not have a default value for `PASSWORD`, and add a `setup` method to handle it.  Something like:

```
def setup
  @password = datastore['PASSWORD']
  @password ||= Rex::Text.rand_text_alpha(16)
end
```

And then just use `@password` anywhere you were previously using `datastore['PASSWORD']`